app/workers/generate_test_applications.rb

```diff
@@ -4,13 +4,21 @@ class GenerateTestApplications
   def perform
     raise 'You can\'t generate test data in production' if HostingEnvironment.production?
 
-    (1..11).each do |i|
-      create_an_application(i)
+    without_slack_message_sending do
+      (1..11).each do |i|
+        create_an_application(i)
+      end
     end
   end
 
 private
 
+  def without_slack_message_sending
+    RequestStore.store[:disable_slack_messages] = true
+    yield
+    RequestStore.store[:disable_slack_messages] = false
+  end
```

Comment:
This speaks to my lack of experience with Rails and maybe threading, but I haven't seen this pattern before.

If I understand correctly, this is setting a global variable, reading it somewhere, and then setting it to something else. I'm not sure I like this approach, as it relies on side-effects, and I have a preference for functional-ish code.

In JS this would look like:

```javascript
global.disableSomething = true;
runMyCode();
global.disableSomething = false;
```

Were there alternatives considered to this approach? 🤔

Reply:
Agreed, it's not ideal.

@duncanjbrown and I discussed a number of approaches. You could for example pass down a `slack: false` variable from the test generator into the services, or do the same with some kind of dependency injection.

Those options probably impact the readability of the code more than the sub-optimal use of globals.

Note that to avoid some of the inherent problems with globals, we've used https://github.com/steveklabnik/request_store. This means the variable won't affect other running processes.